### PR TITLE
Major stable/redis version: remove value inheritance and add optional sentinel support

### LINKED ISSUE #856: [stable/redis] Making chart deployment HA with Sentinels?
Any chance the [stable/redis](https://github.com/helm/charts/tree/master/stable/redis) chart can be refactored to provide genuine HA setup with **Redis Sentinels** (_proper failover and slave promotion to master nodes_)?

Official [stable/redis-ha](https://github.com/helm/charts/tree/master/stable/redis-ha) chart isn't working properly, it's not very production-friendly and PRs are merged at quite slow pace.

I think plenty of people would prefer Bitnami-built Redis HA chart with customizable persistence, unlike [incubator/redis-cache](https://github.com/helm/charts/tree/master/incubator/redis-cache) chart which does Sentinels but not persistence too (_this is a big deal for lots of deployments and companies_). This chart could be used as a starting poing I guess.

Thanks in advance and looking forward to hearing opinions.

### PR DESCRIPTION
**Pull request by javsalgar**
#### What this PR does / why we need it:
This PR was already reviewed in different local requests. Several major changes are proposed:

- Remove master/slave value inheritance (https://github.com/javsalgar/charts/pull/1):  in the long term, the current master-slave value inherintance approach lacked transparency and was difficult to maintain.
- Change slaves from deployments to statefulsets (https://github.com/javsalgar/charts/pull/2): in the current approach (slaves without persistence), in the case of failover, the new slave would have to sync all the data from the very beginning. In terms of performance and I/O consumption, this is problematic. If we have persistence in the slaves, according to the redis documentation (https://redis.io/topics/replication):

> When the link between the master and the slave breaks, for network issues or because a timeout is sensed in the master or the slave, the slave reconnects and attempts to proceed with a partial resynchronization: it means that it will try to just obtain the part of the stream of commands it missed during the disconnection.

- Add optional redis-sentinel support (https://github.com/javsalgar/charts/pull/3): adds Redis Sentinel support to the current configuration (requested by several users). This causes that, instead of two redis services (master and slave) exposed, we only have one with two ports
    - 6379: Redis read only access
    - 26379: Redis sentinel

 For write operations, the user would need to query sentinel first.

  Note: This is disabled by default. The static master-slave approach is the one enabled by default.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes https://github.com/bitnami/charts/issues/856

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by javsalgar** [comment]
@juan131 @tompizmor @prydonius @desaintmartin

**Comment by carrodher** [comment]
@javsalgar a "How to upgrade to 7.0.0" is welcome in the _README.md_

File: stable/redis/README.md [hunk 1]
```diff
@@ -280,9 +349,69 @@ sysctlImage:
...
+
+In case the current master crashes, the Sentinel containers will elect a new master node.
+
+## Notable changes
+
+### 7.0.0
+In order to improve the performance in case of slave failure, we added persistence to the read-only slaves. That means that we moved from Deployment to StatefulSets. This should not affect upgrades from previous versions of the chart, as the deployments did not contain any persistence at all.
+
+This version also allows enabling Redis Sentinel containers inside of the Redis Pods (feature disabled by default). In case the master crashes, a new Redis node will be elected as master. In order to query the current master (no redis master service is exposed), you need to query first the Sentinel cluster. Find more information [in this section](#master-slave-with-sentinel).
+<<<<<<< HEAD
```
Comment (prydonius) [inline stable/redis/README.md]: merge conflict

**Review by prydonius — COMMENTED** [review]
This looks good, thanks! It would be good to add a section to the readme explaining how to enable and use sentinel

**Comment by javsalgar** [comment]
Thanks @prydonius, in the README file you can find a section regarding enabling sentinel

**Review by cpanato — APPROVED** [review]

**Pull request merged**